Bottom: 566076017a597309f76ac6a3d202206f30b1d8b1
Top:    7a5ef851c7ffa131c02fb5dc6c2d168eb4f55c90
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-10 15:28:15 +0000

Refresh of try-something-with-the-rtrees

---

diff --git a/src/polygon.c b/src/polygon.c
index 71d1b2e..471b91c 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -172,6 +172,7 @@ biggest (POLYAREA * p)
 {
   POLYAREA *n, *top = NULL;
   PLINE *pl;
+  rtree_t *tree;
   double big = -1;
   if (!p)
     return NULL;
@@ -205,8 +206,11 @@ biggest (POLYAREA * p)
   if (top == p)
     return p;
   pl = top->contours;
+  tree = top->contour_tree;
   top->contours = p->contours;
+  top->contour_tree = p->contour_tree;
   p->contours = pl;
+  p->contour_tree = tree;
   assert (pl);
   assert (p->f);
   assert (p->b);
diff --git a/src/polygon1.c b/src/polygon1.c
index 5e166de..2d4fbe9 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -898,7 +898,7 @@ M_POLYAREA_intersect (jmp_buf * e, POLYAREA * afst, POLYAREA * bfst, int add)
 	if (curcA->Flags.status == ISECTED)
 	  {
 	    the_list = add_descriptors (curcA, 'A', the_list);
-	    if (UNLIKELY (the_list) == NULL)
+	    if (UNLIKELY (the_list == NULL))
 	      error (err_no_memory);
 	  }
     }
